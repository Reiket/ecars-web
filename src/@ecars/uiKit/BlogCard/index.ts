import {BlogCardContent} from '@ecars/uiKit/BlogCard/components/BlogCardContent';
import {BlogCardComponent} from '@ecars/uiKit/BlogCard/BlogCardComponent';
import {BlogCardImage} from '@ecars/uiKit/BlogCard/components/BlogCardImage';

export const BlogCard = Object.assign(BlogCardComponent, {
  Content: BlogCardContent,
  Image: BlogCardImage,
});
