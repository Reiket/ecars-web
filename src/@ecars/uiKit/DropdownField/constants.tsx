import type {DropdownOption} from 'ecars-web-lib';

interface TopbarDropdownsConfig {
  label: string;
  options: DropdownOption[];
}

const shipToConfig: DropdownOption[] = [
  {
    value: 'ca',
    name: 'Canada',
    label: (
      <img
        src="./assets/images/flags/canada.svg"
        alt="ca"
      />
    ),
  },
  {
    value: 'us',
    name: 'USA',
    label: (
      <img
        src="./assets/images/flags/usa.svg"
        alt="usa"
      />
    ),
  },
  {
    value: 'eu',
    name: 'Europe',
    label: (
      <img
        src="./assets/images/flags/europe.svg"
        alt="eu"
      />
    ),
  },
];

const currencyConfig: DropdownOption[] = [
  {
    value: 'usd',
    name: 'US Dollar',
  },
  {
    value: 'eur',
    name: 'Euro',
  },
  {
    value: 'gbp',
    name: 'British Pound',
  },
];

export const topbarDropdownsConfig: TopbarDropdownsConfig[] = [
  {
    label: 'Ship to:',
    options: shipToConfig,
  },
  {
    label: 'Currency:',
    options: currencyConfig,
  },
];
