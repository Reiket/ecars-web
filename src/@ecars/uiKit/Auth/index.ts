import {AuthLayout} from '@ecars/uiKit/Auth/components/AuthLayout';
import {AuthPageWrapperHOC} from '@ecars/uiKit/Auth/constants';

export const AuthPage = Object.assign(AuthLayout, {
  Wrapper: AuthPageWrapperHOC,
});