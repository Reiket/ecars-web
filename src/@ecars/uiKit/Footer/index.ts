import {FooterComponent} from '@ecars/uiKit/Footer/FooterComponent';
import {FooterCopyWrapperHOC, FooterWrapperHOC} from '@ecars/uiKit/Footer/constants';
import {FooterCopy} from '@ecars/uiKit/Footer/components/FooterCopy';
import {FooterCopyList} from '@ecars/uiKit/Footer/components/FooterCopyList';
import {FooterColumn} from '@ecars/uiKit/Footer/components/FooterColumn';
import {FooterLogoBlock} from '@ecars/uiKit/Footer/components/FooterLogoBlock';
import {FooterLinkList} from '@ecars/uiKit/Footer/components/FooterLinkList';

export const Footer = Object.assign(FooterComponent, {
  Wrapper: FooterWrapperHOC,
  Column: FooterColumn,
  LogoBlock: FooterLogoBlock,
  LinkList: FooterLinkList,
});

export const Copy = Object.assign(FooterCopy, {
  Wrapper: FooterCopyWrapperHOC,
  List: FooterCopyList,
});
