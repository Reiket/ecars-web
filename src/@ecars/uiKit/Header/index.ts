import {HeaderComponent} from '@ecars/uiKit/Header/HeaderComponent';
import {HeaderNavbarWrapperHOC, HeaderWrapperHOC} from '@ecars/uiKit/Header/constants';
import {HeaderNavbar} from '@ecars/uiKit/Header/components/HeaderNavbar';
import {HeaderUtils} from '@ecars/uiKit/Header/components/HeaderUtils';
import {HeaderBurger} from '@ecars/uiKit/Header/components/HeaderBurger';
import {HeaderNavbarList} from '@ecars/uiKit/Header/components/HeaderNavbarList';

export const Header = Object.assign(HeaderComponent, {
  Wrapper: HeaderWrapperHOC,
  Utils: HeaderUtils,
  Burger: HeaderBurger,
});

export const Navbar = Object.assign(HeaderNavbar, {
  Wrapper: HeaderNavbarWrapperHOC,
  List: HeaderNavbarList,
});
