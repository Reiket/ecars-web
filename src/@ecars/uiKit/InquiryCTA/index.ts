import {InquiryCtaComponent} from '@ecars/uiKit/InquiryCTA/InquiryCtaComponent';
import {InquiryCtaWrapperHOC} from '@ecars/uiKit/InquiryCTA/constants';

export const InquiryCTA = Object.assign(InquiryCtaComponent, {
  Wrapper: InquiryCtaWrapperHOC,
});
