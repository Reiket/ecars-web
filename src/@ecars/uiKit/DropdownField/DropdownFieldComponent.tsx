import type {FC} from 'react';
import type {DropdownOption, DropdownThemeType, ElementProps} from 'ecars-web-lib';
import {Dropdown, useDropdown} from 'ecars-web-lib';
import {DropdownField} from '@ecars/uiKit/DropdownField';

export interface Props extends ElementProps {
  label: string;
  options: DropdownOption[];
  theme: DropdownThemeType;
}

export const DropdownFieldComponent: FC<Props> = ({label, options, block, theme}) => {
  const {isDropdownOpen, handleOpenClick, handleOutside, selectedOption, handleSelect} = useDropdown(options);
  return (
    <DropdownField.Wrapper
      label={label}
      block={block}
    >
      <Dropdown
        theme={theme}
        block={block}
        onClickOutside={handleOutside}
        isOpen={isDropdownOpen}
        handleOpen={handleOpenClick}
        onSelect={handleSelect}
        selectedOption={selectedOption}
        options={options}
      />
    </DropdownField.Wrapper>
  );
};
