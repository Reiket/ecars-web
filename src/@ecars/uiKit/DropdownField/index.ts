import {DropdownFieldComponent} from '@ecars/uiKit/DropdownField/DropdownFieldComponent';
import {DropdownFieldWrapper} from '@ecars/uiKit/DropdownField/components/DropdownFieldWrapper';

export const DropdownField = Object.assign(DropdownFieldComponent, {
  Wrapper: DropdownFieldWrapper,
});
