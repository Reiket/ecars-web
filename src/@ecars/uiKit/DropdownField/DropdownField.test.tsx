import {DropdownField} from '@ecars/uiKit/DropdownField';
import {render, screen} from '@testing-library/react';
import type {DropdownOption} from 'ecars-web-lib';

const options: DropdownOption[] = [
  {name: 'Option 1', value: '1'},
  {name: 'Option 2', value: '2'},
];

describe('DropdownField Component', () => {
  test('render component correctly', () => {
    const mockLabel = 'Test Label';
    const {container} = render(
      <DropdownField
        theme="dark"
        options={options}
        label={mockLabel}
      />,
    );
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
    expect(screen.getByText(mockLabel)).toBeInTheDocument();
  });
});
