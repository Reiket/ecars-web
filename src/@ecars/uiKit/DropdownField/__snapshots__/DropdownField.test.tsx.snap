// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DropdownField Component > render component correctly 1`] = `
<div>
  <div
    class="dropdown-field"
  >
    <p
      class="dropdown-field__label"
    >
      Test Label
    </p>
    <div
      class="dropdown-menu dropdown-menu--dark"
      data-testid="dropdownTestId"
    >
      <div
        class="dropdown-menu__block"
      >
        <button
          class="dropdown-menu__category"
        >
          Option 1
          <svg
            class="dropdown-menu__arrow"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
            />
          </svg>
        </button>
      </div>
      <ul
        class="dropdown-menu__list"
      >
        <li
          class="dropdown-menu__option _selected-option"
          data-value="1"
        >
           
          Option 1
        </li>
        <li
          class="dropdown-menu__option"
          data-value="2"
        >
           
          Option 2
        </li>
      </ul>
    </div>
  </div>
</div>
`;
